### DISCUSSION TITLE
Is it possible to migrate sites in "Portal Management" to "Power Pages Management"?

### DISCUSSION THREAD

**Discussion by aka963** [microsoft/powerplatform-build-tools#643] [Q&A] [answered]
Hi,

I came across a situation where pac cli commands are unable to do the migration.

I have 2 environments, Dev and Uat. The both the environment was created in the first week of February 2023.

The development activities started immediately, the sites were created in Dev environment. It was getting created under "PORTAL MANAGEMENT".

Mean while Uat environment was idle till now, no sites were created. Now when we create sites in Uat environment, it gets created under "POWER PAGES MANAGEMENT".

Now I need to migrate the site in Dev to Uat, but it is not happening, I followed various articles, including microsoft articles. Below is the brief of approaches I used:

- Solutions approach | https://learn.microsoft.com/en-us/power-pages/configure/power-pages-solutions
    - The sites does not appear when I go for "Add existing > Site"
- Configuration Migration Tool | https://learn.microsoft.com/en-us/power-platform/admin/manage-configuration-data
    - On Dev I created the site using Starter layout 1 template, not able to identify which schema file represents it (https://learn.microsoft.com/en-us/power-pages/admin/migrate-site-configuration?tabs=CMT#transfer-website-metadata). Please let me know if you have any idea
- Power Platform CLI (v2.0.11) | https://learn.microsoft.com/en-us/power-pages/admin/migrate-site-configuration?tabs=CLI
    - Used command: pac paportal download --path [PATH] --webSiteId [WEBSITEID] -mv 1, for downloading the site from Dev. Downloaded successfully
    - Used command: pac paportal upload --path [PATH] -mv 1, for uploading the site to Uat. Upload unsuccessful, ended with many similar kind of messages "XRM Network error: The entity with a name = 'adx_webfile' with namemapping = 'Logical' was not found in the MetadataCache."
    - Used command: pac paportal upload --path [PATH] -mv 2, for uploading the site to Uat. Partial Success, ended with messages "Message: The attachment is either not a valid type or is too large. It cannot be uploaded or downloaded.". I had to go to "Power Pages Management" and delete the partially created site. Already the attachment size was increased to max

After a lot of trials and errors I came across the above commands which were near to success but not entirely successful.

So before I move further can you please let me know whether it is possible to migrate sites in "Portal Management" to "Power Pages Management"? 

Or 

Should I manually transfer the files from dev site "Portal Management" to uat site "Power Pages Management"?

Or

Any other approach?

Please provide the details of the suggestion or recommendation you give.

Thank you

**Comment by aka963** [comment, marked as answer]
Yes it is possible. I found out that there was some issues with type of file, had to remove or rename it by providing valid type.

**Reply by CatfishLum** [reply]
Hi, what do you mean wrong type? I'm curious because I need to do the same thing and migrate a Portal to a Power Page

**Reply by aka963** [reply]
In my case the issue was in the "web-files" folder. It was having some files which was not having extension like .css or .js and some files did not had there references in the required manifest files. So I had to provide correct extension and for the files that had reference missing I had to remove it and then re-created the file manually and pasted the code content from the file that was removed.

**Reply by CatfishLum** [reply]
Ahhh yeah I see. So you did the CLI command, it created the records and the site (or did you create it manually through Power Pages?), and then you had to correct the Web Files extensions?

Thanks a lot for the tips. I'll try this Monday.

**Reply by aka963** [reply]
I removed the files with issue and then manually added it using "Power Platform Tools" extension in VS Code. I used CLI commands:

To download : pac paportal download --path [PATH] --webSiteId [WEBSITEID] -mv 1
To upload: pac paportal upload --path [PATH] -mv 2

Make sure you connected to the right environment while downloading and uploading.

**Reply by amitjoshi438** [reply]
@aka963 how you created the new files? Did you use the create functionality provided by "Power Platform Tools" extension in VSCode?

**Reply by aka963** [reply]
Yes
